pkg/descheduler/strategies/toomanyrestarts.go

```diff
@@ -30,7 +30,8 @@ import (
 // There are too many cases leading this issue: Volume mount failed, app error due to nodes' different settings.
 // As of now, this strategy won't evict daemonsets, mirror pods, critical pods and pods with local storages.
 func RemovePodsHavingTooManyRestarts(client clientset.Interface, strategy api.DeschedulerStrategy, nodes []*v1.Node, evictLocalStoragePods bool, podEvictor *evictions.PodEvictor) {
-	if strategy.Params.PodsHavingTooManyRestarts.PodRestartThreshold < 1 {
+	if strategy.Params.PodsHavingTooManyRestarts == nil || strategy.Params.PodsHavingTooManyRestarts.PodRestartThreshold < 1 {
```

Comment:
Just a nit. You might `strategyParams := strategy.Params.PodsHavingTooManyRestarts` and do:
```
if strategyParams == nil || strategyParams.PodRestartThreshold < 1 {
```
Also, the same for remaining three occurrences in `for _, node := range nodes {` loop.

Reply:
Hm, what does that benefit besides one less `.`?

Reply:
Nothing, just more shiny code :) Feel free to ignore it.